src/snyk/snykOss/services/vulnerabilityCount/ossVulnerabilityCountService.ts

```diff
@@ -0,0 +1,228 @@
+import npmValidPackageName from 'validate-npm-package-name';
+import { TextDocument } from 'vscode';
+import { IConfiguration } from '../../../common/configuration/configuration';
+import { ILog } from '../../../common/logger/interfaces';
+import { Disposable } from '../../../common/vscode/types';
+import { IVSCodeWindow } from '../../../common/vscode/window';
+import { IVSCodeWorkspace } from '../../../common/vscode/workspace';
+import { HTML_FILE_REGEX, JAVASCRIPT_FILE_REGEX, TYPESCRIPT_FILE_REGEX } from '../../constants/languageRegex';
+import nativeModules from '../../constants/nativeModules';
+import { VulnerabilityCountEmitter, VulnerabilityCountEvents } from '../../vulnerabilityCountEmitter';
+import { OssService } from '../ossService';
+import { ImportedModule } from './importedModule';
+import { ModuleParserProvider } from './parsers/moduleParserProvider';
+import { ModuleVulnerabilityCountProvider } from './vulnerabilityCountProvider';
+
+export enum SupportedLanguage {
+  TypeScript,
+  JavaScript,
+  HTML,
+  PJSON,
+}
+
+export class OssVulnerabilityCountService {
+  protected disposables: Disposable[] = [];
+
+  private emitters = new Map<string, VulnerabilityCountEmitter>();
+
+  constructor(
+    private readonly workspace: IVSCodeWorkspace,
+    private readonly window: IVSCodeWindow,
+    private readonly config: IConfiguration,
+    private readonly vulnerabilityCountProvider: ModuleVulnerabilityCountProvider,
+    private readonly ossService: OssService,
+    private readonly logger: ILog,
+  ) {}
+
+  activate(): boolean {
+    if (this.config.isDevelopment) {
+      // Do not activate service in production until we have feature flag story.
```

Comment:
Sure that's right? Shouldn't it return true, else false? If it's right, maybe the method name should be negated.

Reply:
What's the reasoning of returning true? Not sure I get it. False is returned since service is not getting activated. Purely for unit testing purposes.

Reply:
My bad, fixed :)